application/src/main/java/run/halo/app/security/device/DeviceEndpoint.java

```diff
@@ -0,0 +1,163 @@
+package run.halo.app.security.device;
+
+import static io.swagger.v3.oas.annotations.media.Schema.RequiredMode.REQUIRED;
+import static org.springdoc.core.fn.builders.apiresponse.Builder.responseBuilder;
+import static org.springdoc.core.fn.builders.parameter.Builder.parameterBuilder;
+import static run.halo.app.extension.index.query.QueryFactory.equal;
+
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.security.Principal;
+import java.util.Comparator;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+import org.springdoc.webflux.core.fn.SpringdocRouteBuilder;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.session.Session;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.RouterFunction;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+import run.halo.app.core.extension.Device;
+import run.halo.app.core.extension.endpoint.CustomEndpoint;
+import run.halo.app.extension.GroupVersion;
+import run.halo.app.extension.ListOptions;
+import run.halo.app.extension.ReactiveExtensionClient;
+import run.halo.app.extension.router.selector.FieldSelector;
+import run.halo.app.security.session.ReactiveIndexedSessionRepository;
+
+/**
+ * Device endpoint for user profile,every user can only manage their own devices.
+ *
+ * @author guqing
+ * @since 2.17.0
+ */
+@Component
+@RequiredArgsConstructor
+public class DeviceEndpoint implements CustomEndpoint {
+    private final ReactiveExtensionClient client;
+    private final ReactiveIndexedSessionRepository<?> sessionRepository;
+    private final DeviceService deviceService;
+
+    @Override
+    public RouterFunction<ServerResponse> endpoint() {
+        final var tag = "uc.api.security.halo.run/v1alpha1/Device";
```

Comment:
```suggestion
        final var tag = "DeviceV1alpha1Uc";
```

修改完成之后需要重新生成一下 OpenAPI 文档和 api-client。